test/wh_test_clientserver.c

```diff
@@ -0,0 +1,501 @@
+
+    int counter = 1;
+    char recv_buffer[WH_COMM_MTU] = {0};
+    char send_buffer[WH_COMM_MTU] = {0};
+    uint16_t send_len = 0;
+    uint16_t recv_len = 0;
+
+    uint32_t client_id = 0;
+    uint32_t server_id = 0;
+    uint32_t avail_size = 0;
+    uint32_t reclaim_size = 0;
+    whNvmId avail_objects = 0;
+    whNvmId reclaim_objects = 0;
+
+    /* Check that the server side is ready to recv */
+    WH_TEST_ASSERT_RETURN(WH_ERROR_NOTREADY ==
+            wh_Server_HandleRequestMessage(server));
+
+    for (counter = 0; counter < REPEAT_COUNT; counter++) {
+
+        /* Prepare echo test */
+        sprintf(send_buffer, "Request:%u", counter);
+        send_len = strlen(send_buffer);
+        sprintf(recv_buffer, "NOTHING RECEIVED");
+        recv_len = 0;
+
+        WH_TEST_RETURN_ON_FAIL(
+            wh_Client_EchoRequest(client, send_len, send_buffer));
+#if defined(WH_CFG_TEST_VERBOSE)
+        printf("Client EchoRequest:%d, len:%d, %.*s\n",
+               ret, send_len, send_len, send_buffer);
+#endif
+
+        if (counter == 0) {
+            WH_TEST_ASSERT_RETURN(WH_ERROR_NOTREADY ==
+                                  wh_Client_EchoResponse(
+                                      client, &recv_len, recv_buffer));
+        }
+
+        WH_TEST_RETURN_ON_FAIL(
+            wh_Server_HandleRequestMessage(server));
+
+#if defined(WH_CFG_TEST_VERBOSE)
+        printf("Server HandleRequestMessage:%d\n", ret);
+#endif
+
+        WH_TEST_RETURN_ON_FAIL(
+            wh_Client_EchoResponse(client, &recv_len, recv_buffer));
+
+#if defined(WH_CFG_TEST_VERBOSE)
+        printf("Client EchoResponse:%d, len:%d, %.*s, expected:%.*s\n",
+            ret, recv_len, recv_len, recv_buffer, send_len, send_buffer);
+#endif
+        WH_TEST_ASSERT_RETURN( recv_len == send_len);
+        WH_TEST_ASSERT_RETURN( strncmp(recv_buffer, send_buffer, recv_len) == 0);
+    }
+
+    /* Perform NVM tests */
+    WH_TEST_RETURN_ON_FAIL(
+            wh_Client_NvmInitRequest(client));
+    WH_TEST_RETURN_ON_FAIL(
+            wh_Server_HandleRequestMessage(server));
+    WH_TEST_RETURN_ON_FAIL(
+            wh_Client_NvmInitResponse(client, &client_id, &server_id));
+
+#if defined(WH_CFG_TEST_VERBOSE)
+        printf("Client NvmInitResponse:%d, clientid:%d serverid:%d\n",
+            ret, client_id, server_id);
+#endif
+
+    WH_TEST_RETURN_ON_FAIL(
+            wh_Client_NvmGetAvailableRequest(client));
+    WH_TEST_RETURN_ON_FAIL(
+            wh_Server_HandleRequestMessage(server));
+    WH_TEST_RETURN_ON_FAIL(
+            wh_Client_NvmGetAvailableResponse(client,
+                    &avail_size, &avail_objects,
+                    &reclaim_size, &reclaim_objects));
+#if defined(WH_CFG_TEST_VERBOSE)
+        printf("Client NvmGetAvailableResponse:%d, avail_size:%d avail_objects:%d, reclaim_size:%d reclaim_objects:%d\n",
+            ret, avail_size, avail_objects, reclaim_size, reclaim_objects);
+#endif
+
+#if 0
```

Comment:
is this dead code going to be removed in your next commit w/ client NVM stuff? If not we should remove

Reply:
Going to be turned on.